src/event/Event.php

```diff
@@ -51,6 +51,9 @@ public function call() : void{
 			//this exception will be caught by the parent event call if all else fails
 			throw new \RuntimeException("Recursive event call detected (reached max depth of " . self::MAX_EVENT_CALL_DEPTH . " calls)");
 		}
+		if($this instanceof AsyncEvent){
+			throw new \InvalidArgumentException("Cannot call async event synchronously, use callAsync() instead");
+		}
```

Comment:
Again, we're still sharing architecture here. This function should simply not exist for an async event.